### PR TITLE
feat(ReportDetails): RHICOMPL-1969 - Include systems not reporting

### PR DESCRIPTION
**Pull request by bastilian**
This adds systems not reporting to the table and chart on the report details page and PDF.

Screenshot 2022-01-20 at 09 57 56

⚠️ It is behind a feature flag and can be enabled via appending `?systemsNotReporting=enabled` to the report details URL.

## Secure Coding Practices Checklist GitHub Link
- https://github.com/RedHatInsights/secure-coding-checklist

## Secure Coding Checklist
- [ ] Input Validation
- [ ] Output Encoding
- [ ] Authentication and Password Management
- [ ] Session Management
- [ ] Access Control
- [ ] Cryptographic Practices
- [ ] Error Handling and Logging
- [ ] Data Protection
- [ ] Communication Security
- [ ] System Configuration
- [ ] Database Security
- [ ] File Management
- [ ] Memory Management
- [x] General Coding Practices

### CODE REVIEW

File: src/SmartComponents/ReportDetails/Components/hooks/useLegendData.js [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+  const notReportingHostCount =
+    totalHostCount - unsupportedHostCount - testResultHostCount;
+
+  return [
+    {
+      name:
+        donutValues[0].y +
+        ' ' +
+        pluralize(donutValues[0].y, 'system') +
+        ' compliant',
```
Comment (skateman) [inline src/SmartComponents/ReportDetails/Components/hooks/useLegendData.js:22]: Can't we make these more readable with string interpolation?

**Review by skateman — CHANGES_REQUESTED** [review]

File: src/SmartComponents/ReportDetails/Components/hooks/useLegendData.js (hunk 1 above)
Comment (bastilian) [reply src/SmartComponents/ReportDetails/Components/hooks/useLegendData.js]: Updated.

Usually these kind of nitpicks don't really justify a "requested change". Nitpick sure, but it shouldn't block a PR from being merged.

**Comment by skateman** [comment]
Looks good, but I think I found another bug that will be more visible after merging this:
Screenshot from 2022-01-26 08-36-43
I have 2 systems assigned to the policy, but I only see one here. I guess because only one is reporting. cc @LightOfHeaven1994

File: src/SmartComponents/ReportDetails/Components/hooks/useLegendData.js [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+    ...(notReportingHostCount > 0
+      ? [
+          {
+            name: `${notReportingHostCount} ${pluralize(
+              notReportingHostCount,
+              'system'
+            )} not reporting`,
+            tooltip:
+              notReportingHostCount +
+              ' systems are not reporting scan results. This may be because the system is disconnected, or the insights-client is not properly configured to use Compliance.',
```
Comment (skateman) [inline src/SmartComponents/ReportDetails/Components/hooks/useLegendData.js:47]: Maybe we need singular/plural version here? system vs systems
Not sure if we do it across the product though, so just asking...

**Comment by bastilian** [comment]
@skateman That is no bug, it is not intented to show those systems there at the moment. We can check with UX/PM, but the mockups do not suggest anything to implement on the Reports page for now.

File: src/SmartComponents/ReportDetails/Components/hooks/useLegendData.js (hunk 2 above)
Comment (bastilian) [reply src/SmartComponents/ReportDetails/Components/hooks/useLegendData.js]: Nice catch! I added pluralising here.

**Review by skateman — APPROVED** [review]
The Seal of Approval

**Pull request merged**